cnd/src/config/settings.rs

```diff
@@ -483,13 +489,73 @@ mod tests {
 
         assert_that(&settings)
             .is_ok()
-            .map(|settings| &settings.lnd)
-            .is_equal_to(Some(Lnd {
-                macaroon: None,
-                http_rpc_socket: Socket {
-                    address: IpAddr::V4(Ipv4Addr::UNSPECIFIED),
-                    port: 443,
-                },
-            }))
+            .map(|settings| &settings.lightning)
+            .is_equal_to(Lightning {
+                network: bitcoin::Network::Bitcoin,
+                lnd: Some(Lnd {
+                    rest_api_socket: Some(SocketAddr::new(IpAddr::V4(Ipv4Addr::UNSPECIFIED), 8080)),
+                    dir: Some(PathBuf::from("~/.lnd")),
+                }),
+            })
+    }
+
+    #[test]
+    fn lnd_dir_defaults() {
```

Comment:
I agree they are verbose.  Each test however only has three statements, set up the `File` object, set up the `Settings` object, and assert they match.  I wanted to test all the combinations of the nested options because the code to write them, although quite simple,  took some thought to get the logic 'right'.

Reply:
Fair enough :)